Veir/Dialects/LLVM/Byte/Basic.lean

```diff
@@ -0,0 +1,41 @@
+module
+
+public import Veir.Dialects.LLVM.Int.Basic
+
+namespace Veir.Dialects.LLVM
+
+namespace Byte
+
+public section
+
+structure Byte (w : Nat) where
+  val : BitVec w
+  poison : BitVec w
+deriving DecidableEq
+
+def xor {w : Nat} (x y : Byte w) : Byte w :=
+  ⟨x.val ^^^ y.val, x.poison ||| y.poison⟩
+
+def or {w : Nat} (x y : Byte w) : Byte w :=
+  ⟨x.val ||| y.val, x.poison ||| y.poison⟩
+
+def and {w : Nat} (x y : Byte w) : Byte w :=
+  ⟨x.val &&& y.val, x.poison ||| y.poison⟩
+
+open Int
+
+def toInt {w : Nat} (x : Byte w) : Int w :=
+  if x.poison = 0 then
+    .val x.val
+  else
+    .poison
```

Comment:
This looks weird to me. I would pass the value the integer should be in case the byte is poison. Here you are relying on the internals of the byte implementation which would prevent some reasoning.